provider/aws.go

```diff
@@ -85,6 +108,10 @@ func (p *AWSProvider) Records(zone string) ([]*endpoint.Endpoint, error) {
 			for _, rr := range r.ResourceRecords {
 				endpoints = append(endpoints, endpoint.NewEndpoint(aws.StringValue(r.Name), aws.StringValue(rr.Value)))
 			}
+
+			if r.AliasTarget != nil {
+				endpoints = append(endpoints, endpoint.NewEndpoint(aws.StringValue(r.Name), aws.StringValue(r.AliasTarget.DNSName)))
+			}
```

Comment:
nit: This case is not exercised in the tests.

Could be solved by adding a relevant endpoint in `TestAWSRecords()` :)